src/pages/Swap/LimitOrderBox/components/LimitOrderForm/LimitOrderForm.tsx

```diff
@@ -0,0 +1,513 @@
+          setFormattedLimitPrice,
+          formattedBuyAmount,
+          setFormattedBuyAmount,
+          expiresIn,
+          setExpiresIn,
+          expiresInUnit,
+          setExpiresInUnit,
+          setToMarket,
+        }}
+      >
+        <AutoColumn gap="12px">
+          <AutoColumn gap="3px">
+            <CurrencyInputPanel
+              id="limit-order-box-sell-currency"
+              currency={sellTokenAmount.currency}
+              onCurrencySelect={currency => {
+                const prevSellTokenAmount = sellTokenAmount
+                const amountWei = prevSellTokenAmount?.raw
+                  ? prevSellTokenAmount.raw.toString()
+                  : formattedSellAmount
+                  ? parseUnits(formattedSellAmount, prevSellTokenAmount?.currency?.decimals).toString()
+                  : '0' // use 0 if no buy currency amount is set
+                handleSellCurrencyAmountChange({ currency, amountWei, amountFormatted: formattedSellAmount })
+              }}
+              value={formattedSellAmount}
+              onUserInput={handleInputOnChange}
+              onMax={() => {
+                if (!sellCurrencyMaxAmount) return
+                handleSellCurrencyAmountChange({
+                  currency: sellCurrencyMaxAmount?.currency as Token,
+                  amountWei: sellCurrencyMaxAmount?.raw.toString(),
+                  amountFormatted: sellCurrencyMaxAmount.toSignificant(),
+                })
+              }}
+              maxAmount={sellCurrencyMaxAmount}
+              fiatValue={fiatValueInput}
+              isFallbackFiatValue={isFallbackFiatValueInput}
+              showNativeCurrency={false}
+              currencyOmitList={[buyTokenAmount.currency.address!]}
+            />
+            <SwapTokens swapTokens={handleSwapTokens} loading={loading} />
+            <CurrencyInputPanel
+              id="limit-order-box-buy-currency"
+              currency={buyTokenAmount?.currency}
+              onCurrencySelect={currency => {
+                const prevBuyTokenAmount = buyTokenAmount
+                const amountWei = prevBuyTokenAmount?.raw
+                  ? prevBuyTokenAmount.raw.toString()
+                  : formattedBuyAmount
+                  ? parseUnits(formattedBuyAmount, prevBuyTokenAmount?.currency?.decimals).toString()
+                  : '0' // use 0 if no buy currency amount is set
+                handleBuyCurrencyAmountChange({ currency, amountWei, amountFormatted: formattedBuyAmount })
+              }}
+              value={formattedBuyAmount}
+              onUserInput={formattedValue => {
+                const amountFormatted = formattedValue.trim() === '' ? '0' : formattedValue
+                const amountWei = parseUnits(formattedValue, buyTokenAmount.currency.decimals).toString()
+                handleBuyCurrencyAmountChange({
+                  currency: buyTokenAmount.currency,
+                  amountWei,
+                  amountFormatted,
```

Comment:
This code seems duplicated with `handleInputOnChange`. Only a few variables change.